packages/trimerge-sync-indexed-db/src/testLib/BasicDiffer.ts

```diff
@@ -0,0 +1,57 @@
+import type { Differ, MergeStateFn } from 'trimerge-sync';
+
+// Basic trimerge function that merges values, strings, and objects
+import {
+  combineMergers,
+  trimergeEquality,
+  trimergeObject,
+  trimergeString,
+} from 'trimerge';
+import { create, Delta } from 'jsondiffpatch';
+import { produce } from 'immer';
+import { computeRef as computeShaRef } from 'trimerge-sync-hash';
+
+const trimergeObjects = combineMergers(
+  trimergeEquality,
+  trimergeString,
+  trimergeObject,
+);
+export const merge: MergeStateFn<any, any> = (base, left, right) => ({
+  value: trimergeObjects(base?.value, left.value, right.value),
+  editMetadata: {
+    ref: `(${left.ref}+${right.ref})`,
+    message: `merge`,
+  },
+});
+export const jdp = create({ textDiff: { minLength: 20 } });
+
+export function patch<T>(base: T, delta: Delta | undefined): T {
+  if (delta === undefined) {
+    return base;
+  }
+  return produce(base, (draft) => jdp.patch(draft, delta));
```

Comment:
Have you verified that jdp.patch isn't making a deep copy?

Reply:
yea, hit lots of early bugs because I didn't realize it mutates :) they do offer a deep clone function, but this seems more efficient. at some point might be worth writing a version of jdp patch that does what immer does because we have all the context